### DISCUSSION TITLE
Find an icon's UI label or friendly descriptor

### DISCUSSION THREAD

**Discussion by JJSlote** [freeplane/freeplane#1626] [Help] [answered]
Hi - I've been developing a listview in Autohotkey that parses a map's XML and lists the nodes in sortable columns. It's similar to the _Find and Replace_ window on Ctrl+Shift+F but with additional columns such as attributes, node depths, child counts, descendant counts, anything useful.    

Icons, selected via the Structured Icon Toolbar, work well as tags. I'm happy to show 'em as words rather than images. Within the map's XML, these icons are stored as internal identifiers. But the user is likely to have chosen the icon via a semantic description in the UI that is not retained in the map's XML. Thus for example:

_Excellent_ is stored as _bookmark_
_Dangerous_ is stored as _clanbomber_
_Triangular Ruler_ is stored as _emoji-1F4D0_

**_Question:_** is there a table a script could interrogate to map the internal icon name back to the friendly description by which it was selected? That's what I'd prefer to show in my listview. 

Workarounds there are of course. With just a few icons in use at present, I can maintain my own table as an array pair or string within the script. Also, user icons are displayed and selected by their filenames, so I can make some PNGs and choose meaningful names for them.  

Still interested in a native solution if anybody knows of one.

Thanks
Jerome

**Comment by euu2021** [comment, marked as answer]
I don't know a solution for your situation, but I can say that the [Developer Tools Add-on](https://docs.freeplane.org/getting-started/Add-ons_(install).html#-developer-tools) has an inspector of menu items that shows information about the icons. For example, when inspecting one icon, it returns this kind of information:

```
Menu item: grinning face
Key: IconAction.emoji-1F600
Icon Key: IconAction.emoji-1F600.icon
Command line option (copied to clipboard): -XIconAction.emoji-1F600
Path: //main_menu/insert//icons/menu_iconByCategory//icons/icons/icons/IconAction.emoji-1F600
Shortcut: 
```

So, one idea is to take a look at the methods used in the Developer Tools add-on, to understand how it works.

**Reply by JJSlote** [reply]
Thanks, euu, you've put me on the right track. Chances are excellent that as the Developer Tools add-on can interrogate a single menu item, it can be looped to generate a list. Your link also pointed me to the User Icons add-on, which copies larger versions of the built-in icons into the user icons subfolder, where I can rename as needed. So I'm good to go, with a way to get under the hood as well. Thanks again!!

**Comment by euu2021** [comment]
Maybe, it's possible to use `getTranslatedDescription()`. Take a look at this script: https://github.com/macmarrum/freeplane-scripting/blob/main/src/scripts/iconMetricsTable.groovy
